From usawashi16@yahoo.co.jp Sat Apr 27 13:53:18 2019
From: 宇砂ワシ <usawashi16@yahoo.co.jp>
Message-ID: <557105870.1148761.1556372794832.JavaMail.yahoo@jws702108.mail.ssk.yahoo.co.jp>
Subject: histedit: Fix infinite loop when using 'fc -s'
Date: Sat, 27 Apr 2019 22:46:34 +0900 (JST)

When compiling with libedit, he.num may somehow overrun last, causing
an infinite loop when using fc -s.
This mild change to the check plugs it.

Bug-Debian: https://bugs.debian.org/928072

--- a/src/histedit.c
+++ b/src/histedit.c
@@ -388,7 +388,7 @@
 		 * At end?  (if we were to lose last, we'd sure be
 		 * messed up).
 		 */
-		if (he.num == last)
+		if (he.num >= last)
 			break;
 	}
 	if (editor) {
